### ISSUE TITLE
Quest Announce is Broken

### ISSUE THREAD

**Issue by Gogo1951** [Questie/Questie#3492]
I suspect you guys are doing this on purpose, but in the even that you just didn't test any of this before releasing it... quest announce is off by default for new installs of Questie.

Also, when you select "announce on quest completed" it also announce on quest turn in. The double announce is just annoying, was never the original intent of the Quest Announce feature.

**Comment by BreakBB** [comment]
@lbuchholz79 might be able to share insight on this as #3273 was the PR bringing this new feature in 😁 

We will also look into it to get this fixed soon. Thanks for letting us know @Gogo1951

**Comment by Muehe** [comment]
As TheCrux already explained the default was changed by a recent PR, but this also ties into issues #3458 and #3220 so be aware the default might stay a while.

**Comment by lbuchholz79** [comment]
There is definitely an issue here. This is what happened:
Before the PR there was only one option "Announce objective completion to party members" (`Questie.db.char.questAnnounce`) that could be switched on or off to announce objective progress to the group.

With the PR that setting has been migrated to the new "Announce quest updates via chat" (`Questie.db.char.questAnnounceChannel`) channel option, defaulting to "group" when it was set earlier. New options for accepted, abandoned or completed quest events have been added. These defaulted to off by intend, rendering request #3458 partially useless except for an account-wide toggle. 

Unfortunately the old objective completion option slipped through and neither was migrated to the proper new option nor was it a condition in such cases. This is definitely something that need to be rectified. 

I also noticed there is no option to disable quest starter items. I think it makes sense adding an option for these as well while I'm at it.

**Comment by lbuchholz79** [comment]
Secondly, having account-wide or char-wide settings is not part of #3273 and that's a separate decision @BreakBB needs to make as it touches all setting dialogues.

Thirdly, despite the default being "off", if people enable these options deliberately, spamming of Questie can occur and is a(n intended) side effect. However, hiding or filtering such messages as proposed in #3220 is a totally different topic.

Final issue about double announce: I'll have a look into this. If it is an issue, it more related to #3111 when the event system is raising or handling events in an improper order. On the other hand, there is also a message cache implemented, that skips messages already being sent. This can be tested by accepting and abandoning the same quest in a row.

**Comment by Muehe** [comment]
Yes, they are technically separate issues, but might influence a decision regarding defaults here, so I wanted to link them just for context. Looking forward to to your findings.

FYI if you want to introduce and/or change stuff specifically when people update, and only once, you can add it to [Modules/Migration.lua](https://github.com/Questie/Questie/blob/e0585d3b04c24560c4983bc76fc9833cee4f11b2/Modules/Migration.lua).

**Comment by Muehe** [comment]
P.S.: My current thinking is approach it in this order by the way:

1. Change announce defaults (or at least some of them) back to on
2. Add "Questie ShutUp!" option (defaulting to off)
3. Move settings to (account-wide) profile system

Main reason being the last step will probably necessitate a full settings reset to defaults for everybody, so everything else should ideally be taken care off by then.

**Comment by lbuchholz79** [comment]
I agree, but we need to agree on sensible defaults first.
As some users seem offended just by broadcasting objective progress, setting **all** options to "on" does not seem like the best idea to me, 

Even if so, the "Shut Up" option should be implemented **first**.

Regarding account-wide settings: do people **only** want announcement options to be account wide or **all** Questie settings? In the former case, there might be a simpler solution that just rewrite the whole settings dialogue. In later case, Ace3 should provide all what's needed, but having people use it correctly mostly is a challenge on its own.

**Comment by Muehe** [comment]
I'm totally fine with defaulting just the completion announcement to on, and leaving the others off. This gives a sensible discoverability of the feature while being less intrusive then it could be, especially if the ShutUp option is added. If people in your group want you to turn the others on they will tell you I guess.

> Even if so, the "Shut Up" option should be implemented first.

Order on the first two is not that important IMHO, they just both need to be done before 3, regardless whether the settings are moved or refactored to profiles.

But that settings discussion proper should be had on #3458, I have already given a rationale on why I think profiles would be a better than just making settings account wide. Situation right now is we have both apparently, some settings are per character, some per account, and it's not indicated anywhere. Which is basically the worst of all possible solutions. :sweat_smile: "Historical reasons"™

**Comment by lbuchholz79** [comment]
@Gogo1951 I was unable to reproduce the double announce issue, unless you mean the yellow Blizzard chat messages and the Quest messages. 
Only possible scenario is the QUEST_TURNED_IN event is fire twice for the same quest. Could you please provide more details and which options you have turned on (auto completion for instance)?

**Comment by BreakBB** [comment]
> P.S.: My current thinking is approach it in this order by the way:
> 
>     1. Change announce defaults (or at least some of them) back to on
> 
>     2. Add "Questie ShutUp!" option (defaulting to off)
> 
>     3. Move settings to (account-wide) profile system
> 
> 
> Main reason being the last step will probably necessitate a full settings reset to defaults for everybody, so everything else should ideally be taken care off by then.

I agree to this procedure. If "Questie ShutUp" is added first that is also fine for me 👍🏻 

Also the "account wide" vs "per character" settings are indeed historically based. I am quite sure Questie initially had all settings account wide, but requests came in for specific settings to be useful to change them per character, so we changed these.

A profile system is A) the defacto standard for bigger addons and B) an intuitive solution to decide which settings are active on which character.

**Comment by Gogo1951** [comment]
> @Gogo1951 I was unable to reproduce the double announce issue, unless you mean the yellow Blizzard chat messages and the Questie messages. Only possible scenario is the QUEST_TURNED_IN event is fired twice for the same quest. Could you please provide more details and which options you have turned on (auto completion for instance)?

I have auto-complete on, and auto-accept. 

When I complete the objectives, I get a "Quest Complete Message" and when I turn the quest in, it fires again showing the same message. As far as I know, I don't have any other add-ons that would interfere with Questing, maybe Double Wide Quest Log, but I can't see that triggering a quest complete event.

**Comment by Gogo1951** [comment]
> As some users seem offended just by broadcasting objective progress, setting **all** options to "on" does not seem like the best idea to me...

You gotta ignore the people who get offended, mate. They're just trolls. Have a vision for what you want, and run with it. Ignore the haters. 

The quest announce feature was designed to help spread awareness for Queste at the start of Classic. Questie came from private servers, and so a lot of private server players knew about it, but most of the people coming back for Classic weren't private server players and we needed a way to help ensure everyone knew there were questing tools out there.

It was passive advertising for us. Helped ensure Questie was the default questing tool.

With SoM, a number of other tools started popping up. Things that helped people quest faster. I had said before that Questie was good as a general purpose tool, but if it doesn't adapt it's going to lose users. And that sucks. Nobody wants to watch a 99% user base dwindle.

My opinion... quest announce was done and fine. Energy should be put into real features that will help Questie retain users. 

Something like a feature to help people sort out junk items they don't need any more after turn in a quest, or something to help them only see "Kill" quests (or a step up only Kill + Guaranteed Loot Quests)... 

Having this many people mucking with Quest announce... meh, fine, you do you. I'm not on the project any more. But if I had this much time with devs, I'd focus them elsewhere. (=

**Comment by lbuchholz79** [comment]
> I agree to this procedure. If "Questie ShutUp" is added first that is also fine for me 👍🏻
> 
> Also the "account wide" vs "per character" settings are indeed historically based. I am quite sure Questie initially had all settings account wide, but requests came in for specific settings to be useful to change them per character, so we changed these.
> 
> A profile system is A) the defacto standard for bigger addons and B) an intuitive solution to decide which settings are active on which character.

I think nobody argues against Ace3 profile system, as it has everything one needs and is very simple to use, at least for devs. But I can tell you from my own little addon, that some users have their problems getting the hang of the various profiles and don't know if they are account, realm or char specific. Also the default Ace3 profile UI is not very intuitive either.

In my personal opinion, account-wide announce options run contrary to the initial purpose. On my leveling alt, I might want an announce, but not on my raid or banking char. 

That being said, I'd rather have a toggle on every group/tab if all settings there should be applied account, realm or char wide. I just dont know if somehting like `Questie.db.$storageMode.announceOption` would work dynamically.

**Comment by lbuchholz79** [comment]
> I have auto-complete on, and auto-accept.
> 
> When I complete the objectives, I get a "Quest Complete Message" and when I turn the quest in, it fires again showing the same message. As far as I know, I don't have any other add-ons that would interfere with Questing, maybe Double Wide Quest Log, but I can't see that triggering a quest complete event.

I thought as much, and activated these options y'day as well. Also running Double Wide Quest Log and Questie Event logging along with it. No turn-in event fired twice nor did the message. Maybe it was a specific quests?

**Comment by lbuchholz79** [comment]
Oh, I almost forgot. I'm on holiday next week, so no new input from my side meanwhile. Also I provided PR #3504 to fix the defaults and the unconditional announces on objectives progress.

**Comment by Gogo1951** [comment]
> Also the "account wide" vs "per character" settings are indeed historically based. I am quite sure Questie initially had all settings account wide, but requests came in for specific settings to be useful to change them per character, so we changed these.

Bite the bullet and switch to standard profiles... like every other addon uses at this point. (=

This was one of the goals of the Options UX Updates as well, to switch off the really janky UX around character or account-based settings.

**Comment by Gogo1951** [comment]
![image](https://user-images.githubusercontent.com/55365231/169732350-906c8af0-5465-4ca0-b938-4b029e8716cf.png)

Still doing the double announce, once when you complete. Once at turn in. 

Note how it announced the quest AFTER picking up the follow-up quest.

**Comment by Laumesis** [comment]
That's debatable if that is a double announce. It depends heavily how you have your chat (windows) set up. 😛

**Comment by Gogo1951** [comment]
@Laumesis 

It announced twice. Once when the last objective was completed. That's good.

Once at turn-in. That's bad.

Has nothing to do with chat windows as far as I know. It's firing twice.

**Comment by Laumesis** [comment]
@Gogo1951  Well, I guess I didn't and do not understand what your issue is as your screen shots didn't show it at all.

**Comment by Gogo1951** [comment]
@Laumesis the "complete" message announces twice.

![Pasted_Image_5_23_22__6_39_PM](https://user-images.githubusercontent.com/55365231/169921184-c6cf9eea-21ca-434a-9277-ba9cbc13fa3a.png)

The screenshot showed it completing the quest (cuz you'd have to in order to get the follow up), then picking up the follow up, then announcing the completion of the quest a second time. It didn't show the fist announce, that happened when I picked up the last sack of loot. Then I had to run all the way back to the quest giver.

It already announced it when I finished the quest, lots of running between those points, on a high-pop server the chat messages were too long to capture them both. Does the bug make sense? The goal isn't to announce when all objectives are complete, and then announce again at turn in. There should not be an announce at turn in. That's the bug, the 2nd announce at turn in.

**Comment by Laumesis** [comment]
@Gogo1951 Your screen shot has only one Questie message:
> ![image](https://user-images.githubusercontent.com/55365231/169732350-906c8af0-5465-4ca0-b938-4b029e8716cf.png)
So I have no idea what you are talking about 🤷 .

**Comment by Gogo1951** [comment]
@BreakBB You marked this closed, and @Laumesis doesn't seem to understand my bug report. Do you think the double announce is fixed?

**Comment by Laumesis** [comment]
@Gogo1951 Your latest explanation even is contradiction.
> It announced when the quest was done, and then again when the quest was turned in.

and

> First Quest Turned In. (Implied)
> First Quest Announced. (The Bug).

**Comment by Gogo1951** [comment]
@Laumesis Go do a quest. It'll announce when you complete the quest. Then it'll announce the same thing again when you turn it in. It should not announce on turn in. I don't know what else to say here. I feel like you're just trolling me.

**Comment by Laumesis** [comment]
@Gogo1951 Dunno who is trolling who 🤷 ?
I still don't know what your issue is. Have you tried this?
![image](https://user-images.githubusercontent.com/75958501/169926013-e457508d-05a1-4c4e-86e9-224aeb3e37fd.png)

**Comment by Gogo1951** [comment]
@Laumesis So you're saying it's supposed to announce when the quest is "turned in" and you call that "quest completed"?

And you think "Objective completed" should be when the quest is done? The way the feature used to work?

Even though "Objectie" (note the singular) doesn't actually announce when legs of the quest are done, it only announces when all are done? Fuck the UX here is just garbage. I have no idea what these boxes are supoosed to do by reading this, and no clue what the intent was other than to waste time.

I'm out. This is really bad UX.

**Comment by Laumesis** [comment]
@Gogo1951 I'm glad to hear I could help you.

**Comment by Muehe** [comment]
@lbuchholz79 FYI :point_up_2: 

My workaround is prefixing the number with issue/PR when I use [keywords](https://docs.github.com/en/enterprise-server@3.4/get-started/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests), e.g. "partially fixes issue #3492".

**Comment by Gogo1951** [comment]
@Muehe Double Announce is "fixed"... but the UX and copy on these options is still atrocious.

No clue why anyone would want to have options here, it's just on or off. If it's on, it should do completed (the way it used to), and great... add in quest failed. Nobody is going to want to just announce quest fails, right? So like why give that as an option. 

Better solution would just be to have "Quest Announce" and one simple checkbox, and then have it output completed, like it did before, and just bolt in "Failed" into that. 

![image](https://user-images.githubusercontent.com/55365231/169932684-120225d4-faf8-4c64-a03c-50034f678bdb.png)

It's using "Quest Complete" when it really means "Quest Turned In" and we shouldn't give that as an option. The words are off, it's confusing. You know my take on this, all easily solved with wireframes first. Mock it up, get the words right, make sure the UX makes sense.

Quest announce was designed to help you know when your friends were good to stop killing boars for you. You're being run through things... and cool, you can stop killing things. There's actionable info there.

Just spamming out, "I turned in a Quest!" oof. That's horrible. Shouldn't exist. I'd turn on Questie Shut Up if I ever saw that. Ha.

There's no user story here, there's no "As a [User Type] I want to do [some action] in order to [reap some reward]." And as such, there's on way to test any objectives, or ROI on this. 

Quest announce was designed to help market Questie too, but look... I really like the Questie Shut Up option for users who already downloaded Questie, that's a nice add. Meh on everything else. It's poor execution.

And... a few patches ago, all of the announce options were off by default. I had a strong objection to that. See the point above about passive marketing. You're about to have a massive influx of new users to Classic from the Wrath Era... be aggressive about getting them on Quesite. 

You think Quesite is hard to maintain now... imagine when market share shrinks and it's just an also-ran? Like that's what scares me about bad decisions like this. There's no goal in mind, no rewards, no way to track if it's a win. We call these "toy train" issues at work... something for someone to putter around with. Less time on those the better.

**Comment by Muehe** [comment]
PRs are always welcome. :wave:

**Comment by Muehe** [comment]
> Double Announce is "fixed"...

Oh damn, I think you misunderstood @Gogo1951. I just re-opened this issue. It was accidentally closed by merging #3504, because that used a keyword in the PR description.

**Comment by BreakBB** [comment]
So imo what is left to do for this issue is to rename the existing "Quest completed" setting to "Quest turned in". That should make it more clear what this option is doing.

And we have to open up a new issue for the profile system we want to add.

**Comment by lbuchholz79** [comment]
@Gogo1951: There is a "user story" behind in #3273. Just because you don't see the benefit or it's not phrased as a perfret IREB phrase doesn't make a less a user request. I'm not saying my request stands for hundreds other people, but just outright rejecting it is a very stubborn and short-sighted view on things.
I agree naming is bad, but at at least its consistent across all other Questie options and tooltips. "Objectives" as a term seems to be very old indeed, maybe it even falls back to Blizzard themselves.
Default (with the fix) now should behave as before the PR. **If you don't like the feature, keep it off**! But hammering on all buttons and wondering that it does the things it's supposed to do, is just ... maeh. Feel free to come up with some nice UX designs and namings, but ranting at everything is just not gonnan help. (Beside I don't think Questie is a commercial product that needs fighting over users and marketshare to make "profit".)

@BreakBB @Muehe: Sorry for the auto-closing, i don't work with Github that often and wasn't familiar with keywords.

@BreakBB You're totally correct, too many issues addressed and errant discussions in this one. I think this issue should be closed as well.

**Comment by Gogo1951** [comment]
Good software takes a village. UX Designers, Devs, Testers, PMs. It takes Translators, and community managers, and marketing. It's a lot more than just devs. But you guys are shit to everyone who doesn't write code. 

This "PRs are welcome" echos that. It's just a shit comment, designed to say, "If you don't dev, you're nothing." And that's the short sighted part. Same as the Discord roles. Same as removing me as a contributor from Curseforge the instant you could, even though there's an "inactive" role that you've given others. So fucking petty.

Questie may not be commercial, but it's not a train set in your basement either -- there are probably a million+ users. It's a project I love, and it hurts watching you guys neglect it and then drive it in the ground; no evidence of any strategic vision going on here. Just a bunch of guys with duct tape... It sucks.

Look, the UX on the Quest Announce refactor is dogshit. You can close this, I agree isn't not productive. 

Whatever happened to the planned alpha / beta channels we had talked about? What ever happened to the testing network I had started to build... where we had at least one person dedicated to every class from each faction? Just got ignored and tossed, I imagine. Design > Build > Test > Launch. Core concept of every software project. The only part of that you guys want to do is build... and the product suffers.

**Comment by Muehe** [comment]
> But you guys are shit to everyone who doesn't write code.

I beg to differ.

> So fucking petty.

Says the guy who has been leaving passive-aggressive or outright insulting comments in basically every issue he is in for the past weeks. Look you know why I just said "PRs are welcome"? Because I'm not reading through another text wall that is 50% a rant about how we are the worst people to ever be born and another 40% your irrelevant musings about project leadership.

I have seen you make more complex PRs than changing back a default. But instead here you are, crying wolf. So I posit that you don't actually want to help. You just want to air out your frustration. Maybe both, but I for one am sick of digging through your bullshit to see if there *might* be a kernel of truth hidden inside.

> Look, the UX on the Quest Announce refactor is dogshit.

Let's agree to disagree.

> You can close this, I agree isn't not productive.

Fine by me. :shrug: As you well know part of your complaints are discussed in another issue anyway.
